routing/joint_index.cpp

```diff
@@ -17,6 +17,21 @@ void JointIndex::AppendToJoint(Joint::Id jointId, RoadPoint const & rp)
   m_dynamicJoints[jointId].AddPoint(rp);
 }
 
+void JointIndex::FindPointsWithCommonFeature(Joint::Id jointId0, Joint::Id jointId1,
+                                             vector<pair<RoadPoint, RoadPoint>> & result) const
+{
+  result.clear();
+  ForEachPoint(jointId0, [&](RoadPoint const & rp0) {
+    ForEachPoint(jointId1, [&](RoadPoint const & rp1) {
+      if (rp0.GetFeatureId() == rp1.GetFeatureId())
+        result.emplace_back(rp0, rp1);
+    });
+  });
+
+  if (result.empty())
```

Comment:
What's wrong? Why do we need to throw an exception here? In case of zero intersection, the |result| will be empty. Let the caller to decide what to do if two joints don't share any feature.